pkg/apis/servicecatalog/types.go

```diff
@@ -118,6 +118,13 @@ type ServiceBrokerStatus struct {
 	// was last processed by the controller. The reconciled generation is updated
 	// even if the controller failed to process the spec.
 	ReconciledGeneration int64
+
+	// RetryFailureTime is the time after which reconciliation retries should no
+	// longer be attempted on the ServiceBrokerSpec. After this time, the
+	// ServiceBroker is considered Failed.
+	// If RetryFailureTime is nil, then reconciliation of the spec is not in
+	// a retry cycle.
+	RetryFailureTime *metav1.Time `json:"retryFailureTime,omitempty"`
```

Comment:
Could we come up with a name that would imply something that this is the last time? When I read this name (say in the list of the object)I don't get the sense that this represents that after this time gets to fail. Also, isn't there a timestamp when this object was last modified and couldn't we just use that during reconcile to decide if we should move the object to failed state? I guess I'm just wondering if we really need to jot this down to persistent store for each of these object types or can we just derive this value?
If we have a way to know when the reconcile started (or when the object was last modified -> hence kicking off a reconcile) seems like this is information that we could derive.
I think for a user it might be more interesting to know when a reconcile started in any case, rather than how long it has left before it's considered failed.

Reply:
@vaikas-google I don't think we have a timestamp for when the resource spec was last modified. We have a `LastTransitionTime` on a resource condition, but this signifies to when that condition was updated in the resource's status, and is constantly overwritten in a failure loop.

How about `RetryTimeoutTime`?

Reply:
I still think that having a time when the spec was last modified (whether it exists today or we add that in the status?) instead of a very specific field. It would be more useful to the customer to see when say the spec was changed rather than writing a value down that is very specific to only when we transition to a failure state.

Reply:
Especially since this value is easily computed based on the knobs or servicebroker specific value.

Reply:
So, I like the OperationStartTime (which was basically what I was trying to say here), I'm still not sure we need to jot down the deadlinetime since it always seems to be computed. Adding fields is easy, removing, not so much.

Reply:
> Adding fields is easy, removing, not so much.

True.  If we can all agree on OperationStartTime we can make it better later after more pressing things are done.

Thoughts everyone?

Reply:
`OperationStartTime` sounds good to me.

Reply:
I have removed `RetryFailureTime` and replaced it with `Operation StartTime`